confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java

```diff
@@ -775,6 +778,74 @@ public DataSet showRegion(GetRegionInfoListPlan getRegionsinfoReq) {
     }
   }
 
+  @Override
+  public DataSet showDataNodes() {
+    TSStatus status = confirmLeader();
+    GetRegionInfoListPlan getRegionsinfoReq = new GetRegionInfoListPlan();
+    DataNodeInfosResp dataNodeInfosResp = new DataNodeInfosResp();
+    if (status.getCode() != TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+      dataNodeInfosResp.setStatus(status);
+      return dataNodeInfosResp;
+    }
+    List<TDataNodesInfo> dataNodesInfoList = nodeManager.getOnlineDataNodesInfoList();
+    RegionInfoListResp regionsInfoDataSet =
+        (RegionInfoListResp) partitionManager.getRegionInfoList(getRegionsinfoReq);
+
+    // Map<DataNodeId, DataRegionNum>
+    Map<Integer, AtomicInteger> dataRegionNumMap = new HashMap<>();
+    // Map<DataNodeId, SchemaRegionNum>
+    Map<Integer, AtomicInteger> schemaRegionNumMap = new HashMap<>();
+
+    if (regionsInfoDataSet.getRegionInfoList() != null) {
+      List<TRegionInfo> regionInfoList = regionsInfoDataSet.getRegionInfoList();
+
+      regionInfoList.forEach(
+          (regionInfo) -> {
+            int dataNodeId = regionInfo.getDataNodeId();
+            int regionTypeValue = regionInfo.getConsensusGroupId().getType().getValue();
+
+            if (dataRegionNumMap.get(dataNodeId) != null) {
+              dataRegionNumMap.put(
+                  dataNodeId,
+                  dataRegionNumMap.get(dataNodeId).addAndGet(regionTypeValue)
+                          == TConsensusGroupType.DataRegion.getValue()
+                      ? new AtomicInteger(1)
+                      : new AtomicInteger(0));
+              schemaRegionNumMap.put(
+                  dataNodeId,
+                  schemaRegionNumMap.get(dataNodeId).addAndGet(regionTypeValue)
+                          == TConsensusGroupType.SchemaRegion.getValue()
+                      ? new AtomicInteger(1)
+                      : new AtomicInteger(0));
+            } else {
+              dataRegionNumMap.put(
+                  dataNodeId,
+                  regionTypeValue == TConsensusGroupType.DataRegion.getValue()
+                      ? new AtomicInteger(1)
+                      : new AtomicInteger(0));
+              schemaRegionNumMap.put(
+                  dataNodeId,
+                  regionTypeValue == TConsensusGroupType.SchemaRegion.getValue()
+                      ? new AtomicInteger(1)
+                      : new AtomicInteger(0));
+            }
```

Comment:
Fisrt judge the regionType, then:

`map.computeIfAbset(dataNodeId, empty -> new AtomicInteger(0)).getAndIncrement`

Reply:
fixed~